Bookstore: initialize project with components

VicPeralta opened the pull request:
# Bookstore: initialize the project with components

In this first milestone:

- Used React Router.
- Used React components.
- No linter errors.
- Correct GitFlow.
- Repo documented in a professional way.
- Initialized a CRA app.
- A directory `components` was created to host all components.
- Two routes were created (Books and Categories)
- Books is the default view, it contains a list of books (just one book) with buttons for Comments, Edit, and Remove with no functionality.
- Books view contains a form for adding a book, no functionality provided yet.
- Categories view just displays a button.
- No major styling was used.

VicPeralta committed:
Add main components

Install react-router.
Create components for Book, Books, InputForm, NavBar and Categories.
Create pages for Books and Categories

modified:   package-lock.json
modified:   package.json
modified:   src/App.js
new file:   src/components/Book/Book.css
new file:   src/components/Book/Book.js
new file:   src/components/Books/Books.css
new file:   src/components/Books/Books.js
new file:   src/components/Categories/Categories.js
new file:   src/components/Input-Form/Input-Form.css
new file:   src/components/Input-Form/Input-Form.js
new file:   src/components/NavBar/NavBar.css
new file:   src/components/NavBar/NavBar.js
modified:   src/index.js
deleted:    src/logo.svg
new file:   src/pages/books-page.js
new file:   src/pages/categories-page.js
deleted:    src/reportWebVitals.js

VicPeralta committed:
Update Readme file

new file:   LICENSE.md
modified:   README.md
new file:   desktop.png
new file:   mobile.png
modified:   src/components/Book/Book.css
new file:   tablet.png

FrederickMih commented on src/components/Book/Book.js:
```diff
@@ -0,0 +1,16 @@
+import './Book.css';
+
+const Book = () => (
+  <div className="book">
+    <p>Action</p>
+    <h3>The Hunger Games</h3>
+    <p>Suzanne Collins</p>
```
- [ ] Nice job for displaying the title and author of the book however requirements say the `Book` should take the book data as a prop and renders the title and the author of the book. Please let's considered this requirement too.

FrederickMih commented on src/components/Input-Form/Input-Form.js:
```diff
@@ -0,0 +1,17 @@
+      Author
+      <input type="text" name="author" id="author" required />
+    </label>
+    <input type="submit" />
```
- [ ] Please can we make this a button?. The requirements of this project stipulate that this should be a button.

FrederickMih commented on src/components/NavBar/NavBar.js:
```diff
@@ -0,0 +1,41 @@
+import { NavLink } from 'react-router-dom';
+import './NavBar.css';
+
+const links = [
+  {
+    id: 1,
+    path: '/',
+    text: 'BOOKS',
+  },
+  {
+    id: 2,
+    path: '/categories',
+    text: 'CATEGORIES',
+  },
+];
```
- [ ] I found this piece of code interesting. Keep up!

FrederickMih reviewed (changes requested):
Hi @VicPeralta , 

You have done a fantastic job on this project so far!
You are left with little issues that you still need to work on to go to the next project but you are almost there!
![](https://media.giphy.com/media/L422JoNlBBBD42zkQ9/giphy.gif)

# To Highlights :rocket: 

- Followed GitHub flow :heavy_check_mark: 
- All linters checks are passing :heavy_check_mark: 
- Your README file is professional and descriptive. :+1: 


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

VicPeralta committed:
Add button to Categories component

Install prop-types
Pass book information as props

modified:   package.json
modified:   src/components/Book/Book.js
modified:   src/components/Books/Books.js
modified:   src/components/Categories/Categories.js
modified:   src/components/Input-Form/Input-Form.css
modified:   src/components/Input-Form/Input-Form.js
modified:   src/pages/categories-page.js

BrianSammit reviewed (approved):
# STATUS: APPROVED :green_circle: :clap: 

Hello @VicPeralta, 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

![well-1](https://user-images.githubusercontent.com/64173189/128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c.gif)

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
